packages/components/src/ButtonsGroup/ButtonsGroup.js

```diff
@@ -0,0 +1,165 @@
+import { forwardRef, useEffect, useState } from 'react';
+import { createStyleSheet } from '@platzily-ui/styling';
+import { PropTypes } from 'prop-types';
+
+const useStyleSheet = createStyleSheet(
+  (theme, { borderWrapperComponent }) => ({
+    buttonsGroupWrapper: {
+      width: 'auto',
+      height: 'auto',
+      display: 'inline-block',
+      borderStyle: 'solid',
+      borderColor: theme.palette.neutral.secondary,
+      borderWidth: 1,
+      borderRadius: borderWrapperComponent || theme.spacing(),
+    },
+    firstButtonStyle:{
+      borderTopLeftRadius: borderWrapperComponent || theme.spacing(),
+      borderBottomLeftRadius: borderWrapperComponent || theme.spacing(),
+    },
+    lastButtonStyle:{
+      borderTopRightRadius: borderWrapperComponent || theme.spacing(),
+      borderBottomRightRadius: borderWrapperComponent || theme.spacing(),
+    },
+    separationLinesButton: {
+      borderRightStyle: 'solid',
+      borderRightColor: theme.palette.neutral.secondary,
+      borderRightWidth: 1,
+    },
+    buttonsStyles:{
+      padding: '5px 10px',
+    },
+    buttonUnselected: {
+      backgroundColor: theme.palette.neutral.light,
+      color: theme.palette.neutral.dark
+    },
+    buttonSelected: {
+      backgroundColor: theme.palette.primary.main,
+      color: theme.palette.neutral.light
+    },
+  }),
+  { key: 'ButtonsGroup' },
+);
+
+const ButtonsGroup = forwardRef(function ButtonsGroup(props, ref) {
+  const { actions, className, classNameButtons, separationLinesButtonProp, selectedStyles, unselectedStyles, selectedButtonDefault, borderWrapperComponent, ...otherProps } = props;
+  const [state, setState] = useState(actions);
+  const { classes, cx } = useStyleSheet(props);
+  const stateLength = actions.length;
+
+  useEffect(() => {
+    if (selectedButtonDefault <= (state.length - 1)) {
+      setState(state.map((element, indexMap) => (
+        (selectedButtonDefault === indexMap)
+          ? {
```

Comment:
This ternary can be avoided if you return the object this way:

return {
   ...element,
   selected: index === indexMap
}